detox/src/devices/runtime/drivers/ios/SimulatorDriver.js

```diff
@@ -63,6 +63,11 @@ class SimulatorDriver extends IosDriver {
     }
   }
 
+  async resetAppState(bundleId, binaryPath) {
+    await this.uninstallApp(bundleId);
+    await this.installApp(binaryPath);
```

Comment:
I think we should throw exception here (not implemented) and not to fallback. this will avoid usage of the optimized-app-install feature on iOS (not exists).